### PR TITLE
Remove a few more uses of GeometryInfo.

### PR DESCRIPTION
**Pull request by bangerth**
Follow-up to #4424 .

/rebuild

### CODE REVIEW

File: source/geometry_model/chunk.cc [hunk 1]
```diff
@@ -266,21 +266,22 @@ namespace aspect
         // center is less than 1.e-5 of the distance between vertices (as
         // measured by the minimum distance from any of the other vertices
         // to the first vertex), then we call this a horizontal face.
-        constexpr unsigned int n_vertices = GeometryInfo<dim>::vertices_per_face;
-        std::array<double, n_vertices>     distances_to_center;
-        std::array<double, n_vertices - 1> distances_to_first_vertex;
+        constexpr unsigned int max_n_vertices_per_face = (1 << (dim-1));
```
Comment (peterrum) [inline source/geometry_model/chunk.cc:269]: This does not look optimal. We should make this part of `ReferenceCell`!

Comment (bangerth) [reply source/geometry_model/chunk.cc:269]: Didn't expect to get a review from you here :-)

To the point: True. It's the max over the number of vertices per cell for the `dim-1` dimensional reference cells. But I'm not sure where such a function would go. It isn't information that is part of any specific reference cell. I'm also not sure whether there ought to be a member function that gives us this information.

I did think of writing it as
```
  std::max(ReferenceCells::Hexahedron.face(0).vertices_per_cell, 
                ReferenceCells::Tetrahedron.face(0).vertices_per_cell);
```
But this only works for 3d and would need more work to also make work for 2d, and lastly it is not `constexpr`. So I decided that this must be good enough :-)

Comment (peterrum) [reply source/geometry_model/chunk.cc:269]: > To the point: True. It's the max over the number of vertices per cell for the dim-1 dimensional reference cells. But I'm not sure where such a function would go. It isn't information that is part of any specific reference cell. I'm also not sure whether there ought to be a member function that gives us this information.

We will need this info in `RefernceCell` else we won't be able to remove `GeometryInfo`: the `GeometryInfo` functions are used to initialize `std::arrays` and similar data structures at compile time.  Maybe we can think about `static constexpr` functions. But we don't need to spam aspect, we can continue the discussion at the deal.II GitHub page.

Comment (gassmoeller) [reply source/geometry_model/chunk.cc:269]: I agree that this is kind of good enough in terms of functionality, but I am pretty sure 95% of the readers of this code will not understand what is happening in this line. Wouldnt `(dim == 3) ? 4 : 2` be good enough? Or is that not possible for `constexpr`? We only support 2D and 3D cases in ASPECT anyway.

**Review by gassmoeller — COMMENTED** [review]
Thanks for the improvements. I have one question (see above) but otherwise I think this is good to go. But I will also wait to give @peterrum time to reply.

**Comment by gassmoeller** [comment]
It seems the testers think there is a functional difference in the chunk geometry model between the old version and the new one.

File: source/geometry_model/chunk.cc (hunk 1 above)
Comment (bangerth) [reply source/geometry_model/chunk.cc]: That would work just fine :-)

**Comment by bangerth** [comment]
I don't understand the differences in output. Let me remove the geometry part of the patch for the moment and see that the rest of the patch tests correctly. This should make getting it merged easier.

**Review by gassmoeller — APPROVED** [review]

**Pull request merged**